File: src/Construction.Application/Features/Authentication/Models/AuthResponse.cs
namespace Construction.Application.Features.Authentication.Models;

public class AuthResponse
{
    public string AccessToken { get; init; } = null!;

    public DateTime AccessTokenExpiresAt { get; init; }

    public string RefreshToken { get; init; } = null!;

    public DateTime RefreshTokenExpiresAt { get; init; }

    public UserDto User { get; init; } = null!;
}

File: src/construction_mobile/lib/features/auth/data/models/auth_response.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import 'user.dart';

part 'auth_response.freezed.dart';
part 'auth_response.g.dart';

/// Mirrors the API's `AuthResponse` returned by login and refresh.
@freezed
abstract class AuthResponse with _$AuthResponse {
  const factory AuthResponse({
    required String accessToken,
    required DateTime accessTokenExpiresAt,
    required String refreshToken,
    required DateTime refreshTokenExpiresAt,
    required User user,
  }) = _AuthResponse;

  factory AuthResponse.fromJson(Map<String, dynamic> json) =>
      _$AuthResponseFromJson(json);
}
